impl/src/main/java/com/sun/faces/application/view/FaceletViewHandlingStrategy.java

```diff
@@ -439,14 +440,18 @@ public void renderView(FacesContext ctx, UIViewRoot viewToRender) throws IOExcep
 
             //  Don't call startDoc and endDoc on a partial response
             if (ctx.getPartialViewContext().isPartialRequest()) {
+                Map<Object, Object> contextMap = ctx.getAttributes();
+                contextMap.put(ELFlash.DELAYED_END_DOCUMENT, true);
                 viewToRender.encodeAll(ctx);
+                contextMap.remove(ELFlash.DELAYED_END_DOCUMENT);
                 try {
                     ctx.getExternalContext().getFlash().doPostPhaseActions(ctx);
                 } catch (UnsupportedOperationException uoe) {
                     if (LOGGER.isLoggable(FINE)) {
                         LOGGER.fine("ExternalContext.getFlash() throw UnsupportedOperationException -> Flash unavailable");
                     }
                 }
+                ctx.getPartialViewContext().getPartialResponseWriter().endDocument();
```

Comment:
This does not satisfy the existing comment on line 441. One of them should be adjusted.